File: 002/002.pl
sub fib {
    ($a, $b) = (0, 1);
    return sub {
        ($a, $b) = ($b, $a+$b);
        return $a;
    }
}

my ($f, $n, $s) = (fib(), 0, 0);
while (($n = $f->()) < 4000000) {if (!($n & 1)) {$s += $n}}
print $s, "\n";

File: 002/002.c
int fib(){
    static int a = 0, b = 1;
    int c = a;
    a = b;
    b += c;
    return c;
}

int main(){
    int n = 0, s = 0;
    while ((n = fib()) < 4000000) if (!(n & 1)) s += n;
    printf("%d\n", s);
}
